### PR TITLE
Add more tracing to selenium tests, page source and screenshot

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs [hunk 1]
```diff
@@ -68,88 +68,100 @@ public static void SaveScreenshot(this IWebDriver driver, TestContext testContex
...
+
+            Trace.WriteLine("Logging in ... Entering password");
+            driver.WaitForElementToBeVisibleAndEnabled(fields.GetPasswordInputId()).SendKeys(user.GetOrFetchPassword());
+
+            Trace.WriteLine("Logging in ... Clicking next after password");
+            driver.WaitForElementToBeVisibleAndEnabled(By.Id(fields.GetPasswordSignInButtonId())).Click();
+
+            if (user.HomeUPN.Contains("outlook.com"))
+            {
+                Trace.WriteLine("Logging in ... clicking accept prompts for outlook.com MSA user");
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs]: >outlook.com MSA user [](start = 76, length = 20)

can you use the actual value here, like user.HomeUPN or something?

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿using Microsoft.Identity.Client;
 using Microsoft.Identity.Client.Extensibility;
 using Microsoft.Identity.Client.UI;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs]: license info

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs [hunk 3]
```diff
@@ -105,35 +107,38 @@ private IWebDriver InitDriverAndGoToUrl(string url)
...
+
+
+                    // Run the TCP listener and the selenium automation in parallel
                     var seleniumAutomationTask = Task.Run(() =>
                     {
                         _seleniumAutomationLogic(driver);
                     });
 
-                    await Task.WhenAll(seleniumAutomationTask, listenForAuthCodeTask).ConfigureAwait(false);
+                    //await Task.WhenAll(seleniumAutomationTask, listenForAuthCodeTask).ConfigureAwait(false);
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs]: >// [](start = 20, length = 2)

why is this commented?

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs [hunk 4]
```diff
@@ -27,7 +27,7 @@ public static IWebDriver CreateDefaultWebDriver()
 
             // ~2x faster, no visual rendering
             // remove when debugging to see the UI automation
-            options.AddArguments("headless");
+            //options.AddArguments("headless");
```
Comment (henrik-me) [inline tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs]: I suppose the comment should say, remove // from the line below to see the UI? Also I suspect you didn't intend to change this particular line?

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs [hunk 5]
```diff
@@ -68,88 +68,100 @@ public static void SaveScreenshot(this IWebDriver driver, TestContext testContex
         #endregion
 
         public static IWebElement WaitForElementToBeVisibleAndEnabled(this IWebDriver driver, By by)
-        {
-            WebDriverWait webDriverWait = new WebDriverWait(driver, TimeSpan.FromSeconds(ExplicitTimeoutSeconds));
```
Comment (henrik-me) [inline tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs]: >ExplicitTimeoutSeconds [](start = 89, length = 22)

Any reason to move away from the const?

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs [hunk 6]
```diff
@@ -68,88 +68,100 @@ public static void SaveScreenshot(this IWebDriver driver, TestContext testContex
...
-                    return null;
-                }
-                catch (NoSuchElementException)
-                {
-                    return null;
-                }
-            });
+        {
+            Trace.WriteLine($"[Selenium UI] Waiting for {by.ToString()} to be visible and enabled");
+            var webDriverWait = new WebDriverWait(driver, TimeSpan.FromSeconds(15));
```
Comment (henrik-me) [inline tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs]: >15 [](start = 79, length = 2)

why not the const?

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs (hunk 1 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs]: I think my VS is playing tricks and reformats the line endings. I haven't actually changed anything in this logic.

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs (hunk 3 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs]: Thx for spotting!

File: tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs (hunk 4 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumExtensions.cs]: No, didn't mean to change this. Fixed.

**Pull request merged**